src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -437,102 +271,72 @@ func handleRegistrationFlow(
 	accountDB *accounts.Database,
 	deviceDB *devices.Database,
 ) util.JSONResponse {
-	// TODO: Shared secret registration (create new user scripts)
-	// TODO: Enable registration config flag
-	// TODO: Guest account upgrading
 
-	// TODO: Handle loading of previous session parameters from database.
-	// TODO: Handle mapping registrationRequest parameters into session parameters
-
-	// TODO: email / msisdn auth types.
-
-	if cfg.Matrix.RegistrationDisabled && r.Auth.Type != authtypes.LoginTypeSharedSecret {
-		return util.MessageResponse(http.StatusForbidden, "Registration has been disabled")
+	if res := handleLoginTypeSharedSecret(req, r, cfg, sessionID); res != nil {
```

Comment:
a comment here would be useful: "Before doing the generic user-interactive auth, check some auth mechanisms which are specific to registration".